thunder/torch/__init__.py

```diff
@@ -287,6 +287,30 @@ def to(
     return a
 
 
+@torchsymbol(torch.Tensor.cuda, is_method=True)
+def cuda(a: TensorProxy, /, device=None, non_blocking: bool = False, memory_format: None | torch.memory_format = None):
+    # Modeled similar to PyTorch:
+    # https://github.com/pytorch/pytorch/blob/e3ac61587aa368c613ef01df1f328a396b64cd5d/tools/autograd/templates/python_variable_methods.cpp#L496-L501
+    # If `device` is None, this function defaults `device` to current CUDA device
+    # and delegates actual data-movement and layout ordering to `Tensor.to`.
+
+    # NOTE: `Tensor.to` doesn't model `non_blocking` currently.
```

Comment:
What about...

```
def cuda(a: TensorProxy, /, device: None | DeviceLike=None, **kwargs):
  if device is None:
    cuda_idx: int = torch.cuda.current_device()
    device: devices.Device = devices.Device(devices.DeviceType.CUDA, cuda_idx)

  return to(a, device=device, **kwargs)
```

and let `to` handle the `non_blocking` error message (or not)?

Reply:
`to` doesn't handle `non_blocking` currently.

https://github.com/Lightning-AI/lightning-thunder/blob/139cc2246b87f3e58d80b13897286a61d55826d8/thunder/torch/__init__.py#L239-L252

And leads to error below which seems to be confusing.
```python
  File "/home/kkalambarkar/lightning-thunder/thunder/core/langctxs.py", line 124, in _fn
    result = fn(*args, **kwargs)
TypeError: to() got an unexpected keyword argument 'non_blocking'
```

Reply:
Good point! Although I suppose in the future we should add the `non_blocking` argument to `thunder.torch.to`, since PyTorch's `Tensor.to` supports it.

Reply:
Have filed an issue to support `non_blocking` in `Tensor.to`. See https://github.com/Lightning-AI/lightning-thunder/issues/197